File: src/main/java/mediathek/gui/actions/ShowFilmStatisticsAction.java
package mediathek.gui.actions;

import mediathek.gui.statistics.FilmStatisticsDialog;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;

public class ShowFilmStatisticsAction extends AbstractAction {
    private final Frame owner;

    public ShowFilmStatisticsAction(@NotNull Frame owner) {
        this.owner = owner;
        putValue(Action.NAME, "Filmlisten-Statistik anzeigen...");
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        var dlg = new FilmStatisticsDialog(owner, this);
        dlg.setVisible(true);
    }
}

File: src/main/kotlin/mediathek/gui/actions/ShowFilmStatisticsAction.kt
package mediathek.gui.actions

import mediathek.filmlisten.FilmCatalog
import mediathek.gui.statistics.FilmStatisticsDialog
import java.awt.Frame
import java.awt.event.ActionEvent
import javax.swing.AbstractAction

class ShowFilmStatisticsAction(
    private val owner: Frame,
    private val filmCatalog: FilmCatalog,
) : AbstractAction() {
    init {
        putValue(NAME, "Filmlisten-Statistik anzeigen...")
    }

    override fun actionPerformed(event: ActionEvent?) {
        FilmStatisticsDialog(owner, filmCatalog, this).isVisible = true
    }
}
